python/cudf/cudf/core/scalar.py

```diff
@@ -54,33 +59,85 @@ def __init__(self, value, dtype=None):
         if isinstance(value, DeviceScalar):
             self._data = value
         else:
-            value = to_cudf_compatible_scalar(value, dtype=dtype)
-            valid = not _is_null_host_scalar(value)
-
-            if dtype is None:
-                if not valid:
-                    raise TypeError(
-                        "dtype required when constructing a null scalar"
-                    )
-                else:
-                    dtype = value.dtype
-            dtype = np.dtype(dtype)
-
-            # temporary
-            dtype = np.dtype('object') if dtype.char == 'U' else dtype
-            self._data = DeviceScalar(value, dtype=dtype)
+            self._host_value, self._host_dtype = self._preprocess_host_value(value, dtype)
+
+    
+    @property
+    def _is_host_value_current(self):
+        return self._host_value is not None
+
+    @property
+    def _is_device_value_current(self):
+        return self._device_value is not None
+
+    @property
+    def get_device_value(self):
```

Comment:
Can we call this `device_value`? A `get_` prefix typically implies a method rather than a `property`.

Reply:
Updated